sw/device/lib/testing/aes_testutils.h

```diff
@@ -29,9 +29,9 @@ inline bool aes_testutils_get_status(dif_aes_t *aes, dif_aes_status_t flag) {
  * @param value The status flag value.
  * @param timeout_usec Timeout in microseconds.
  */
-inline void aes_testutils_wait_for_status(dif_aes_t *aes, dif_aes_status_t flag,
-                                          bool value, uint32_t timeout_usec) {
-  IBEX_SPIN_FOR(aes_testutils_get_status(aes, flag) == value, timeout_usec);
-}
+#define AES_TESTUTILS_WAIT_FOR_STATUS(aes, flag, value, timeout_usec)          \
```

Comment:
Why is this a macro now?

Reply:
It does the same as the original function ( spin until the status register changes to the expected value, or fail the test in case of timeout), but now we get a useful line information where the error happened.

Reply:
Seeing as there are three uses in this PR, I wouldn't bother. Maintaining macros is costly, and this one, as implemented, isn't hygenic (it's missing various "standard" guards on its parameters)... fixing it seems like more work that it's worth.

Reply:
There's more three chip level tests to be implemented for AES that will use this MACRO in case we keep it. If you still think it is not worth th a can revert this modification.

Reply:
If you want to do this, you'll need to make it hygenic, which you can do like so:

```c
#define AES_TESTUTILS_WAIT_FOR_STATUS(aes_, flag_, value_, timeout_usec_) \
  IBEX_SPIN_FOR(aes_testutils_get_status((aes_), (flag_)) == (value_), (timeout_usec_))
```

Note that the do/while is unnecessary, since you're expanding to a single expression. It's only necessary for when you want to expand into a block.

Reply:
Thanks!